src/DIRAC/WorkloadManagementSystem/Agent/ScoutingJobStatusAgent.py

```diff
@@ -0,0 +1,227 @@
+
+            scoutIDdict[scoutID] = scoutStatus
+            if scoutStatus['Status'] == 'NotComplete':
+                self.log.verbose(f"{jobID}: skipping since corresponding scout does not complete yet.")
+                continue
+            else:
+                result = self.__updateJobStatus(jobID, status=scoutStatus['Status'],
+                                                minorstatus=scoutStatus['MinorStatus'],
+                                                appstatus=scoutStatus['appstatus'])
+                if not result['OK']:
+                    self.log.warn(result['Message'])
+
+        self.log.info('final scoutIDdict:%s' % scoutIDdict)
+        return S_OK()
+
+    def __getScoutStatus(self, scoutid):
+        ids = scoutid.split(':')
+        scoutjoblist = list(range(int(ids[0]), int(ids[1]) + 1))
+
+        result = self.jobDB.getJobsAttributes(scoutjoblist, ['Status', 'Site'])
+        if not result['OK']:
+            return result
+
+        donejoblist = []
+        donesitelist = []
+        failedjoblist = []
+        failedsitelist = []
+        stalledjoblist = []
+        scoutjobs = result['Value'].keys()
+        for scoutjob in scoutjobs:
+            status = result['Value'][scoutjob]['Status']
+            site = result['Value'][scoutjob]['Site']
+            jobid = scoutjob
+            if status == JobStatus.DONE:
+                donejoblist.append(jobid)
+                donesitelist.append(site)
+            elif status == JobStatus.FAILED:
+                failedjoblist.append(jobid)
+                failedsitelist.append(site)
+            elif status == JobStatus.STALLED:
+                stalledjoblist.append(jobid)
+
+        if self.criteriaSucceeded > len(scoutjobs):
+            criteriaSucceeded = max(int(len(scoutjobs) * self.criteriaSucceededRate), 1)
+            self.log.verbose(f'criteriaSucceeded = {self.criteriaSucceeded}')
+        else:
+            criteriaSucceeded = self.criteriaSucceeded
+            self.log.debug(f'criteriaSucceeded = {self.criteriaSucceeded}')
+
+        if self.criteriaFailed > len(scoutjobs):
+            criteriaFailed = max(int(len(scoutjobs) * self.criteriaFailedRate), 1)
+            self.log.verbose(f'criteriaFailed = {self.criteriaFailed}')
+        else:
+            criteriaFailed = self.criteriaFailed
+            self.log.debug(f'criteriaFailed = {self.criteriaFailed}')
+
+        if self.criteriaStalled > len(scoutjobs):
+            criteriaStalled = max(int(len(scoutjobs) * self.criteriaStalledRate), 1)
+            self.log.verbose(f'criteriaStalled = {self.criteriaStalled}')
+        else:
+            criteriaStalled = self.criteriaStalled
+            self.log.debug(f'criteriaStalled = {self.criteriaStalled}')
+
+        if len(donejoblist) >= criteriaSucceeded:
+            self.log.verbose(f'Scout (ID = {scoutid}) are done.')
+            scoutStatus = {'Status': 'Checking', 'MinorStatus': 'Scouting', 'appstatus': 'Scout Complete'}
```

Comment:
```suggestion
            scoutStatus = {'Status': JobStatus.CHECKING, 'MinorStatus': 'Scouting', 'appstatus': 'Scout Complete'}
```
and similar in the lines below.